### PR TITLE
Add cache_to option in the build and publish commands

### PR DESCRIPTION
**Pull request by marboledacci**
I'm adding a new parameter to allow cache_to similar to the cache_from option.
I'm also updating the orb and tests to support new Mac Silicon and ARM architectures.

### CODE REVIEW

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/docker?version=dev:8bc687e0ad14591e3ca1460e7bfe94a5875f16db

**Review by david-montano-circleci — APPROVED** [review]

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/docker?version=dev:ac2f5c489351673de1adc381a37fff4a722a40e6

**Comment by orb-publisher** [comment]
Your development orb has been published. It will expire in 30 days.
You can preview what this will look like on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/docker?version=dev:ac2f5c489351673de1adc381a37fff4a722a40e6

**Comment by orb-publisher** [comment]
Your orb has been published to the CircleCI Orb Registry.
You can view your published orb on the CircleCI Orb Registry at the following link: 
https://circleci.com/developer/orbs/orb/circleci/docker?version=2.7.0

File: src/scripts/build.sh [hunk 1]
```diff
@@ -66,6 +66,12 @@ if [ -n "$PARAM_CACHE_FROM" ]; then
   build_args+=("--cache-from=$PARAM_CACHE_FROM")
 fi
 
+if [ -n "$PARAM_CACHE_TO" ]; then
+  docker buildx create --name cache --use
+  docker buildx use cache
+  build_args+=("--cache-to=$PARAM_CACHE_TO" --load)
```
Comment (guillaumelecerf) [inline src/scripts/build.sh:72]: Why is `--load` mandatory? I tested with a `Run with SSH` on CircleCI, and the load seems not necessary?

**Pull request merged**